kitsune/src/service/post.rs

```diff
@@ -357,11 +457,261 @@ impl PostService {
+        let content_lang = match update_post.language {
+            Some(lang) => Language::from_639_1(&lang),
+            None => content.as_ref().map(|c| detect_language(c)),
+        };
+
+        let (mentioned_account_ids, content) = match content.as_ref() {
+            Some(content) => {
+                let resolved = self.post_resolver.resolve(content).await?;
+                (resolved.0, Some(resolved.1))
+            }
+            None => (Vec::new(), None),
+        };
+
+        let link_preview_url = if let (Some(embed_client), Some(content)) =
+            (self.embed_client.as_ref(), content.as_ref())
+        {
+            embed_client
+                .fetch_embed_for_fragment(content)
+                .await?
+                .map(|fragment_embed| fragment_embed.url)
+        } else {
+            None
+        };
+
+        let post = self
+            .db_pool
+            .with_transaction(move |tx| {
+                async move {
+                    let post: Post = diesel::update(posts::table)
+                        .set(PostChangeset {
+                            id: update_post.post_id,
+                            subject: subject.as_deref(),
+                            content: content.as_deref(),
+                            content_lang: content_lang.map(Into::into),
+                            link_preview_url: link_preview_url.as_deref(),
+                            is_sensitive: update_post.sensitive,
+                            updated_at: Timestamp::now_utc(),
+                        })
+                        .returning(Post::as_returning())
+                        .get_result(tx)
+                        .await?;
+
+                    Self::process_mentions(tx, post.id, mentioned_account_ids).await?;
+                    Self::process_media_attachments(tx, post.id, &update_post.media_ids).await?;
+
+                    Ok::<_, Error>(post)
+                }
+                .scope_boxed()
+            })
+            .await?;
+
+        self.job_service
+            .enqueue(
+                Enqueue::builder()
+                    .job(DeliverUpdate {
+                        entity: UpdateEntity::Status,
+                        id: post.id,
+                    })
+                    .build(),
+            )
+            .await?;
+
+        if post.visibility == Visibility::Public || post.visibility == Visibility::Unlisted {
+            self.search_service
+                .update_in_index(post.clone().into())
+                .await?;
+        }
+
+        self.status_event_emitter
+            .emit(PostEvent {
+                r#type: EventType::Update,
+                post_id: post.id,
+            })
+            .await
+            .map_err(Error::Event)?;
+
+        Ok(post)
+    }
+
+    /// Boost a post
+    ///
+    /// # Panics
+    ///
+    /// This should never panic. If it does, create a bug report.
+    pub async fn boost(&self, boost_post: BoostPost) -> Result<Post> {
```

Comment:
We usually use the term "repost" throughout the codebase. "Boost" is a bit Mastodon-specific.

Even Mastodon calls it just "repost" in its data schema IIRC